speechtotextv1/speech_to_text_v1_integration_test.go

```diff
@@ -404,3 +400,39 @@ func TestDeleteLanguageModel(t *testing.T) {
 	)
 	assert.Nil(t, responseErr)
 }
+
+type myCallBack struct {
+	T *testing.T
+}
+
+func (cb myCallBack) OnOpen() {
+}
+func (cb myCallBack) OnClose() {
+}
+func (cb myCallBack) OnData(resp *core.DetailedResponse) {
+	var speechResults speechtotextv1.SpeechRecognitionResults
+	result := resp.GetResult().([]byte)
+	json.Unmarshal(result, &speechResults)
+	assert.NotNil(cb.T, speechResults)
+	assert.NotNil(cb.T, speechResults.Results[0].Alternatives[0].WordConfidence)
+	assert.NotNil(cb.T, speechResults.SpeakerLabels)
+	assert.NotNil(cb.T, speechResults.Results[0].Alternatives[0].Timestamps)
+}
+func (cb myCallBack) OnError(err error) {
+	cb.T.Fail()
+}
+
+func TestRecognizeUsingWebsockets(t *testing.T) {
+	f, _ := os.Open("../resources/audio_example.mp3")
+	callback := myCallBack{T: t}
+	wsListener := speechtotextv1.WebsocketListenerFactory(callback)
```

Comment:
Can we avoid having a factory?

Reply:
The function was mainly to allow users to create the websocketListener without having to manually instantiate and track of all the required components.
Would changing the naming to something like `WebsocketListenerMaker` suffice or is there an implementation change that would be more appropriate?